### PR TITLE
Styling: Enable strict null checks in the styling package

### PR DESCRIPTION
**Pull request by christiango**
#### Description of changes
Enable the Typescript strictNullChecks compiler option in the styling package .

### CODE REVIEW

File: packages/styling/src/interfaces/IFontStyles.ts [hunk 1]
```diff
@@ -4,16 +4,16 @@ import { IRawStyle } from '../interfaces/index';
  * UI Fabric font set.
  */
 export interface IFontStyles {
```
Comment (dzearing) [inline packages/styling/src/interfaces/IFontStyles.ts]: I think that seems fine, but I think you will need to update ITheme to support partial font lists.

Comment (christiango) [inline packages/styling/src/interfaces/IFontStyles.ts]: Does theme ever have any of these values as undefined? Don't we always put a default there?

Comment (dzearing) [inline packages/styling/src/interfaces/IFontStyles.ts]: so loadTheme takes ITheme. It support partials. It should be a partial theme with partial/optional fonts and colors. We mix them on top of the current theme.

```ts
loadTheme({ palette: { themePrimary: 'red' } });
```

Comment (dzearing) [inline packages/styling/src/interfaces/IFontStyles.ts]: The final ITheme probably should have required values, like you have. But we need a way such that we can have a partial of partials? How do we do that? I think using `type` you could define IPartialTheme type, with keyof ITheme optionaled to a partial of the type, maybe.

**Comment by dzearing** [comment]
I'd like to get this merged, but i'm not sure how this will affect loadTheme usage, which should have optionals.

File: packages/styling/src/interfaces/IFontStyles.ts (hunk 1 above)
Comment (christiango) [inline packages/styling/src/interfaces/IFontStyles.ts]: Turns out I don't need this change to turn on strict null checks. Though it is worth following up if we can be more strict on this type.

**Comment by christiango** [comment]
I undid the font styles change, should be good to go

**Pull request merged**